### ISSUE TITLE
generateTrackIndexSequences returns negative values when there are no negative array members

### ISSUE THREAD

**Issue by Nheyer** [vgteam/sequenceTubeMap#93]
https://github.com/vgteam/sequenceTubeMap/blob/713b2d993099c4b11326d45b18cf87a4ce0a8022/src/util/tubemap.js#L963

I have spent the last few dayes trying to figure out why when I get to a specific part of the VG graph that was generated from cactus, and simplified via VG simplify, sequenceTubeMap yealds a blank screen and crashes, it seems that this code here is designed to map the absolute value of the nodemap_value , assuming that a negative key ( that is a key starting with the char '-')  implies a negative node index in the map, but this is NOT always the case, and such a case causes an undefined object to be generated when exicuting [this](https://github.com/vgteam/sequenceTubeMap/blob/713b2d993099c4b11326d45b18cf87a4ce0a8022/src/util/tubemap.js#L420)
when we get to [here](https://github.com/vgteam/sequenceTubeMap/blob/713b2d993099c4b11326d45b18cf87a4ce0a8022/src/util/tubemap.js#L2923) and this leads to an exception being thrown [here](https://github.com/vgteam/sequenceTubeMap/blob/713b2d993099c4b11326d45b18cf87a4ce0a8022/src/util/tubemap.js#L2926) causing rendering to be halted indefinitely.

To sum up I just need to know if having negative node-indexes anywhere is an expected behavior, and if changing the code here will cause  any runtime errors, given that  when I modify it locally everything seems to work

**Comment by Nheyer** [comment]
@adamnovak

**Comment by adamnovak** [comment]
There should be no real zero or negative node IDs in the underlying graph.

However, it does seem like we're using negation of a node ID to mean something, and I have no clue what. Maybe strand?

**Comment by Nheyer** [comment]
Do you have an idea who would, because if so, my simple absolute value may be breaking things, even though it seems to be working.

**Comment by adamnovak** [comment]
@wolfib should know, but I think he's moved on to other projects. Maybe @glennhickey? He apparently last touched a bunch of the vg-side code that talks to the tube map.

**Comment by wolfib** [comment]
I am not sure about the context here, but here is what I remember in general: nodes themselves don't have negative indices, but the sequence of nodes a track passes through can contain negative node indices. This simply means that the track traverses the node with that index in reversed direction (i.e. from right to left in the resulting graph).

**Comment by Nheyer** [comment]
ahh ok @wolfib  if this is what is going on pr #94  should fix the issue I was having properly
